src/main/java/com/iwaconsolti/school/controller/CourseController.java

```diff
@@ -0,0 +1,77 @@
+package com.iwaconsolti.school.controller;
+
+import com.iwaconsolti.school.controller.response.CourseRequest;
+import com.iwaconsolti.school.model.Course;
+import com.iwaconsolti.school.service.SchoolService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/{schoolName}/courses")
+public class CourseController {
+
+    @Autowired
+    @Qualifier("gerardoService")
+    private SchoolService gerardoService;
+
+    @Autowired
+    @Qualifier("zetService")
+    private SchoolService zetService;
+
+    private SchoolService getServiceBySchoolName(String schoolName) {
```

Comment:
This is in the 3 controllers, you can extact to anothe bean that contains this logic. 

Note: Ask me about this when you see me